src/main/java/tb/common/block/BlockBriar.java

```diff
@@ -57,6 +59,33 @@ public void updateTick(World w, int x, int y, int z, Random rnd) {
         }
     }
 
+    public boolean onBlockActivated(World aWorld, int aX, int aY, int aZ, EntityPlayer aPlayer, int aSide, float pX,
+        float pY, float pZ) {
+        int aMeta = aWorld.getBlockMetadata(aX, aY, aZ);
+        boolean isTop = isTopBlock(aMeta);
+        int lowerY = isTop ? aY - 1 : aY;
+        int upperY = isTop ? aY : aY + 1;
+
+        int lowerMeta = aWorld.getBlockMetadata(aX, lowerY, aZ);
+        int upperMeta = aWorld.getBlockMetadata(aX, upperY, aZ);
+
+        if (lowerMeta >= growthStages - 1 && upperMeta >= growthStages - 1 + 8) {
+            int fortune = EnchantmentHelper.getFortuneModifier(aPlayer);
+
+            ArrayList<ItemStack> drops = getDrops(aWorld, aX, lowerY, aZ, upperMeta, fortune);
+            for (ItemStack stack : drops) {
+                dropBlockAsItem(aWorld, aX, lowerY, aZ, stack);
+            }
+
+            aWorld.setBlockMetadataWithNotify(aX, lowerY, aZ, 0, 2);
+            aWorld.setBlockMetadataWithNotify(aX, upperY, aZ, 8, 2);
+
+            return true;
+        }
+
+        return false;
```

Comment:
This logic can be changed some to make it cleaner and reduce nesting:
```suggestion
        int upperY = lowerY + 1;

        int lowerMeta = aWorld.getBlockMetadata(aX, lowerY, aZ);
        int upperMeta = aWorld.getBlockMetadata(aX, upperY, aZ);

        if (lowerMeta < growthStages - 1 || upperMeta < growthStages - 1 + 8) return false;

        int fortune = EnchantmentHelper.getFortuneModifier(aPlayer);
        for (ItemStack drop : getDrops(aWorld, aX, lowerY, aZ, upperMeta, fortune)) {
            dropBlockAsItem(aWorld, aX, lowerY, aZ, drop);
        }

        aWorld.setBlockMetadataWithNotify(aX, lowerY, aZ, 0, 2);
        aWorld.setBlockMetadataWithNotify(aX, upperY, aZ, 8, 2);

        return true;
```